src/mixed_script.rs

```diff
@@ -126,4 +140,36 @@ impl MixedScript for &'_ str {
     fn resolve_script_set(self) -> AugmentedScriptSet {
         self.into()
     }
+
+    fn detect_restriction_level(self) -> RestrictionLevel {
+        use crate::GeneralSecurityProfile;
+        let mut ascii_only = true;
+        let mut set = AugmentedScriptSet::default();
+        let mut exclude_latin_set = AugmentedScriptSet::default();
+        for ch in self.chars() {
+            if !GeneralSecurityProfile::identifier_allowed(ch) {
+                return RestrictionLevel::Unrestricted;
+            }
+            if ch as u32 > 0x7F {
+                ascii_only = false;
+            }
+            let ch_set = ch.into();
+            set = set.intersect(ch_set);
+            if !ch_set.base.contains_script(Script::Latin) {
+                exclude_latin_set.intersect(ch_set);
```

Comment:
`intersect()` does not mutate.

Perhaps we should have an `intersection()` and `intersect_with()` pair of methods on unicode-script

Reply:
oops